frame/system/src/lib.rs

```diff
@@ -935,7 +987,13 @@ impl<T: Trait> Happened<T::AccountId> for CallOnCreatedAccount<T> {
 pub struct CallKillAccount<T>(PhantomData<T>);
 impl<T: Trait> Happened<T::AccountId> for CallKillAccount<T> {
 	fn happened(who: &T::AccountId) {
-		Module::<T>::kill_account(who.clone());
+		if Account::<T>::contains_key(who) {
+			let account = Account::<T>::take(who);
+			if account.refcount > 0 {
+				sp_runtime::print("WARNING: Referenced account deleted. This is probably a bug.");
```

Comment:
Looks important, would use `frame_support::debug` stuff with `target: System` so that it is filter-able.

Reply:
good idea